doc/platform-optimization/README.md

```diff
@@ -0,0 +1,188 @@
+Kubernetes node.
+
+Managing CPUs and memories enables improving AI container performance
+and maintaining predictable response times even under heavy
+load. Reasons for performance improvements include the following.
+
+- Better cache hit ratios in all cache levels.
+- Fewer remote memory accesses.
+- Fewer processes and threads per CPU in the whole system.
+- Disabling CPU hyperthreading on containers that run faster when the
+  other CPU thread is idle.
+
+More predictable response times are possible by using dedicated CPUs
+for containers and sets of containers. This ensures that critical
+containers will always have enough compute resources, and that
+resource hungry containers will not be able to hurt all processes in
+the system.
+
+## NRI Plugins
+
+[NRI plugins](https://github.com/containers/nri-plugins) connect to
+the container runtime running on a Kubernetes node. Containerd and
+CRI-O runtimes support NRI plugins.
+
+The NRI plugins project includes two resource policies, balloons and
+topology-aware. They manage allowed CPUs and memories (cpuset.cpus and
+cpuset.mems) of all Kubernetes containers created and running on the
+node.
+
+In this example, we use the balloons policy because it can be tuned
+for certain applications (like RAG pipelines) using even node-specific
+parameters for each container in applications. The topology-aware
+policy, on the other hand, needs no configuration and does CPU
+assignment automatically based on resource requests in containers and
+underlying hardware topology.
+
+## Install
+
+Warning: installing and reconfiguring the balloons policy in can
+change allowed CPUs and memories of already running containers in the
+cluster. This may hurt containers that rely on the number of allowed
+CPUs being static.
+
+Install the balloons policy with helm:
+
+1. Add the NRI plugins repository
+   ```bash
+   helm repo add nri-plugins https://containers.github.io/nri-plugins
+   ```
+
+2. Install the balloons resource policy and patch container runtime's
+   configuration on host in order to switch NRI support on.
+   ```bash
+   helm install balloons nri-plugins/nri-resource-policy-balloons --set patchRuntimeConfig=true
+   ```
+
+Now the balloons policy is managing node resources in the cluster as a
+DaemonSet that communicates with the container runtime on every
+node. You should see `nri-resource-policy-balloons-...` pod running on
+every node.
+
+## Configure
+
+Edit the default balloons policy:
+```bash
+kubectl edit balloonspolicy default
+```
+
+Let us consider isolating AI inference and reranking containers in
+[ChatQnA](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA)
+application's Gaudi accelerated pipeline.
+
+In the
+[manifest](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/kubernetes/manifests/gaudi/chatqna.yaml)
+there are "tgi" and "tei" containers that will need a lot of CPUs.
+
+We will allocate 16 CPUs (8 full CPU cores with two hyperthreads on
```

Comment:
May want to add a description sentence like "We refer to physical cores as CPU Cores or full CPU cores and hyperthreaded cores as virtual cores or vCPUs or just CPUs".

.. (16 physical cores that are hyperthreaded making for 32 vCPUs)


Sometimes we need to turn off hyperthreading ... NRI can detect that.

Reply:
Thanks, added!